src/Api/Api.php

```diff
@@ -540,50 +540,19 @@ public function render(string $path, $method = 'GET', array $requestData = [])
         try {
             $result = $this->call($path, $method, $requestData);
         } catch (Throwable $e) {
-            if (is_a($e, 'Kirby\Exception\Exception') === true) {
-                $result = [
-                    'status' => 'error',
-                    'route'  => ($this->route)? $this->route->pattern() : null
-                ] + $e->toArray();
-            } else {
-                // remove the document root from the file path
-                $file = $e->getFile();
-                if (empty($_SERVER['DOCUMENT_ROOT']) === false) {
-                    $file = ltrim(Str::after($file, $_SERVER['DOCUMENT_ROOT']), '/');
-                }
-
-                $result = [
-                    'status'    => 'error',
-                    'exception' => get_class($e),
-                    'message'   => $e->getMessage(),
-                    'file'      => $file,
-                    'line'      => $e->getLine(),
-                    'code'      => empty($e->getCode()) === false ? $e->getCode() : 500,
-                    'route'     => $this->route ? $this->route->pattern() : null
-                ];
-            }
+            $result = $this->responseForException($e);
         }
 
-        if ($result === null) {
-            $result = [
-                'status'  => 'error',
-                'message' => 'not found',
-                'code'    => 404,
-            ];
-        }
-
-        if ($result === true) {
-            $result = [
-                'status' => 'ok',
-            ];
-        }
-
-        if ($result === false) {
-            $result = [
-                'status'  => 'error',
-                'message' => 'bad request',
-                'code'    => 400,
-            ];
+        switch (true) {
```

Comment:
That `switch` looks a bit weird. Why not `if elseif elseif`?

Reply:
I found this to be quite elegant, because it can be extended so easily. To me switches are always easier to read than endless if elseif clauses, but that might be personal.

Reply:
Unfortunately switches don't do strict comparisons, otherwise it would be even cleaner.

Reply:
As you say, it's pretty subjective. To me the special `switch` looks like a hack. A working one, but still a hack. `if elseif elseif` would be a bit less code overall and harder to screw up (you can't forget a `break`). But you can leave it like this if you want.

Reply:
I gave in and replaced it :)